frontend_tests/casper_tests/12-mention.js

```diff
@@ -0,0 +1,85 @@
+var common = require('../casper_lib/common.js').common;
+
+function enter_mention_in_composer(str, item) {
+    casper.then(function () {
+        casper.test.info('Start typing @all');
+        casper.evaluate(function (str, item) {
+            // Set the value and then send a bogus keyup event to trigger
+            // the typeahead.
+            $('#new_message_content')
+                .focus()
+                .val(str)
+                .trigger($.Event('keyup', { which: 0 }));
+
+            // Trigger the typeahead.
+            // Reaching into the guts of Bootstrap Typeahead like this is not
+            // great, but I found it very hard to do it any other way.
+            var tah = $('#new_message_content').data().typeahead;
+            tah.mouseenter({
+                currentTarget: $('.typeahead:visible li:contains("'+item+'")')[0]
+            });
+            tah.select();
+        }, {str: str, item: item});
+    });
+}
```

Comment:
I think probably it makes sense to refactor out the code that's common to this and the similar function in 02-narrow.js into a new function in frontend_tests/casper_lib/common.js, since I think only the selector and `casper.test.info` line are different; does that seem right @knowcliu?  If so I'd probably split that refactoring out as its own commit in this PR (either a commit first that just refactors from 02-narrow.js to common.js so this commit can just call the function, or a commit at the end that recombines).